filters/tracing/tag_test.go

```diff
@@ -91,19 +115,47 @@ func TestTracingTag(t *testing.T) {
 
 			ti.context.FRequest = ti.context.FRequest.WithContext(opentracing.ContextWithSpan(ti.context.FRequest.Context(), span))
 
-			s := NewTag()
-			f, err := s.CreateFilter([]interface{}{"test_tag", ti.value})
+			f, err := ti.filter.CreateFilter([]interface{}{"test_tag", ti.value})
 			if err != nil {
 				t.Fatal(err)
 			}
 
 			f.Request(ti.context)
+			f.Response(ti.context)
 
 			if got := span.Tag("test_tag"); got != ti.expected {
 				t.Errorf("unexpected tag value '%v' != '%v'", got, ti.expected)
 			}
-
-			f.Response(ti.context)
 		})
 	}
 }
+
+func TestTagFilterIgnoresResponse(t *testing.T) {
```

Comment:
I think this could be a test case in the TestTracingTag
```go
	{
		"tag ignores response",
		NewTag(),
		"${response.header.X-Fallback}",
		&filtertest.Context{
			FRequest: &http.Request{
				Header: http.Header{
					"X-Fallback": []string{"true"},
				},
			},
		},
		"",
	}
```
and likely we need the symmetric that NewTagFromResponse ignores request.

Reply:
Not sure about this. Wouldn't the header in the request only match `"${request.header.X-Fallback}"`, not the provided template for the **response**? So in any case the result would be `nil`, right?

My reasoning was that we need to have a case where a header is only present in the response, not in the request. 

Will add a test for the symmetrical case, though.

Reply:
Sorry, I meant to use FResponse of course, i.e. filter is NewTag but template is `"${response.header.X-Fallback}"` and `X-Fallback` is a response header

Reply:
We can have `request` and `response` fields in the test cases instead of context, create context at the start of the test and set response after f.Request(ctx) but before f.Response(ctx):

```go
ctx := &filtertest.Context{
  FRequest: ti.request,
}
...

f.Request(ctx)

ctx.FResponse = ti.response

f.Response(ctx)
```

Reply:
This works for a test that show `tracingTag` only runs on the request. It does not work for the symmetric case to test that `tracingTagFromResponse` only runs on the response. To prove this, the request in the context has to be changed (header removed) to prove that `tracingTagFromResponse` cannot make use of headers that are only present at request time. It is a bit artificial anyway.

Would you prefer to just have one test according to the method you described, or separate tests as they are now? Or a different change?

Reply:
Alternatively, I could change the test code to reset the context to blank after the request call. But this would also be a bit strange, because in a real scenario the request context would be available also at response time, if I understand it correctly.

Reply:
Lets have a single additional test case then.

Reply:
In the additional test you can also have a real backend httptest.Server to create the response header to test it.